src/main/java/org/springframework/data/couchbase/core/query/Query.java

```diff
@@ -40,6 +53,20 @@ public Query addCriteria(QueryCriteria criteriaDefinition) {
 		return this;
 	}
 
+	public Query setParameters(JsonValue parameters) {
```

Comment:
I think it could be benefitial to have two overloads (jsonObject, jsonarray), very similar to what the sdk has

Reply:
Added. Amend coming...

Reply:
well, this is public API so once have it it will stick - can you do two user-visible overloads with JsonObject and JsonArray that cast down into JsonValue for now?